### PR TITLE
Basic task-based peering working

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Add basic underlying capability to generate fake "dummy" assets for every task in an airflow dag.

This changed our fetching logic to unconditionally fetch and serialize all task infos in the airflow instance. I believe this is inevitable but this will make the serialization snapshot much bigger.

For now I've done this in mostly parallel codepath. Once @dpeng817 's composability changes land we might be able to consolidate these codepaths more and eventually have only one public API. But this lets us make progress for now.

## How I Tested These Changes

Manually ran kitchen sink test.

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24852?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24852** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24852?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24850** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24850?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @schrockn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_defs_data.py [hunk 1]
```diff
@@ -80,6 +84,27 @@ def construct_dag_assets_defs(self) -> Definitions:
             ]
         )
 
+    def construct_automapped_dag_assets_def(self) -> Definitions:
```
Comment (dpeng817) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_defs_data.py:87]: misleading name

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_defs_data.py [hunk 2]
```diff
@@ -80,6 +84,27 @@ def construct_dag_assets_defs(self) -> Definitions:
...
+                    key=key_for_task_asset(dag_data.dag_id, task_id),
+                    deps=[
+                        key_for_task_asset(dag_data.dag_id, upstream_task_id)
+                        for upstream_task_id in upstream_task_ids
+                    ],
+                )
+                for task_id, upstream_task_ids in upstream_deps.items()
+            )
+
+        return Definitions.merge(Definitions(assets=specs), self.construct_dag_assets_defs())
```
Comment (dpeng817) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_defs_data.py:106]: [1] we need to link the leaf tasks to the dag asset that we create

File: examples/experimental/dagster-airlift/dagster_airlift_tests/unit_tests/core_tests/test_load_defs.py [hunk 3]
```diff
@@ -425,3 +426,25 @@ def an_asset() -> None: ...
...
+
+    dag1_task1 = key_for_task_asset("dag1", "task1")
+    dag1_task2 = key_for_task_asset("dag1", "task2")
+    dag1_standalone = key_for_task_asset("dag1", "standalone")
+
+    specs = {spec.key: spec for spec in defs.get_all_asset_specs()}
+
+    assert specs[dag1_task1].deps == []
+    assert specs[dag1_task2].deps == [AssetDep(dag1_task1)]
+    assert specs[dag1_standalone].deps == []
```
Comment (dpeng817) [inline examples/experimental/dagster-airlift/dagster_airlift_tests/unit_tests/core_tests/test_load_defs.py:448]: [1] I'd expect some assertions around the full dag asset as well, namely that it has a dep on all of these

**Review by dpeng817 — CHANGES_REQUESTED** [review]
handling of dag asset deps

**Review by dpeng817 — APPROVED** [review]
looks good outside of dag deps. But that's handled in a downstream

**Comment by schrockn** [comment]
### Merge activity

* **Sep 30, 9:52 AM EDT**: @schrockn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24852).

**Pull request merged**